packages/core/src/components/DataTable/types.tsx

```diff
@@ -0,0 +1,251 @@
+import React from 'react';
+import { SortDirectionType } from 'react-virtualized';
+import { WithStylesProps } from '../../composers/withStyles';
+
+export type RowHeightOptions = 'micro' | 'small' | 'regular' | 'large';
+export type HeightOptions = RowHeightOptions | undefined;
+
+export type SelectedRows = {
+  [key: number]: {
+    status: string;
+    selectedChildren: Set<number>;
+  };
+};
+
+export type ChangeLog = {
+  [key: number]: {
+    [key: string]: {
+      newVal: any;
+    };
+  };
+};
+
+type EditCallback = (
+  row: TableRow,
+  key: string,
+  newVal: any,
+  event: React.SyntheticEvent<EventTarget>,
+) => void;
+
+export type HandleEdit = (
+  row: TableRow,
+  key: string,
+) => (newVal: any, event: React.SyntheticEvent<EventTarget>) => void;
+
+export type HeaderButton = {
+  label: string;
+  onClick: (selectedRows: SelectedRows) => () => void;
+  display?: boolean;
+  displayEditMode?: boolean;
+};
+
+export type DefaultDataTableProps =
```

Comment:
`export type DefaultDataTableProps = keyof DataTableProps`?

Reply:
Oh my god yes, this is so much better.